src/lib/util/opresolv.h

```diff
@@ -50,45 +50,79 @@
 //  TYPE DEFINITIONS
 //**************************************************************************
 
-enum option_type
-{
-	OPTIONTYPE_END,
-	OPTIONTYPE_INT,
-	OPTIONTYPE_STRING,
-	OPTIONTYPE_ENUM_BEGIN,
-	OPTIONTYPE_ENUM_VALUE
-};
-
-struct option_guide
-{
-	enum option_type option_type;
-	int parameter;
-	const char *identifier;
-	const char *display_name;
-};
-
 #define OPTION_GUIDE_START(option_guide_)                                   \
-	const option_guide option_guide_[] =                                \
+	const util::option_guide option_guide_ =										\
 	{
 #define OPTION_GUIDE_END                                                    \
-		{ OPTIONTYPE_END }                                                  \
 	};
 #define OPTION_GUIDE_EXTERN(option_guide_)                                  \
-	extern const option_guide option_guide_[]
+	extern const util::option_guide option_guide_
 #define OPTION_INT(option_char, identifier, display_name)                   \
-		{ OPTIONTYPE_INT, (option_char), (identifier), (display_name) },
+	{ util::option_guide::entry::option_type::INT, (option_char), (identifier), (display_name) },
 #define OPTION_STRING(option_char, identifier, display_name)                \
-	{ OPTIONTYPE_STRING, (option_char), (identifier), (display_name) },
+	{ util::option_guide::entry::option_type::STRING, (option_char), (identifier), (display_name) },
 #define OPTION_ENUM_START(option_char, identifier, display_name)            \
-	{ OPTIONTYPE_ENUM_BEGIN, (option_char), (identifier), (display_name) },
+	{ util::option_guide::entry::option_type::ENUM_BEGIN, (option_char), (identifier), (display_name) },
 #define OPTION_ENUM(value, identifier, display_name)                        \
-	{ OPTIONTYPE_ENUM_VALUE, (value), (identifier), (display_name) },
+	{ util::option_guide::entry::option_type::ENUM_VALUE, (value), (identifier), (display_name) },
 #define OPTION_ENUM_END
 
 namespace util {
+
+// ======================> option_guide
+
+class option_guide
```

Comment:
This thing is not POD because of the vector.  You'll need to make it into a kind of abstract base and and implementation templated on the number of items using std::array for storage, then have a variadic creator function that makes one with the right number of elements depending on number of arguments.